tests/constructors/cached_provider_test.py

```diff
@@ -0,0 +1,58 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+import pytest
+
+from beamlime.constructors.providers import CachedProvider
+
+
+def function_with_unhashable_arguments(_: list) -> None:
+    return
+
+
+def random_number() -> float:
+    import random
+
+    return random.random()
+
+
+def random_number_with_seed(seed: int) -> float:
+    import random
+
+    random.seed(seed)
+    return random.random()
+
+
+def test_cached_provider_without_arguments():
+    cached_random_number = CachedProvider(random_number)
+    assert cached_random_number() == cached_random_number()
+    assert cached_random_number() != random_number()
```

Comment:
Just for my understanding: Is this because of the `maxsize=2`?

Reply:
It's because it always returns the first returned value.
And `cached_random_number` and `random_number` is sharing the same `random` object.
That's why it has another test with `random_number_with_seed`, to initialize random seed between calls.